lib/burnside/convert-expr.cc

```diff
@@ -551,7 +558,32 @@ class ExprLowering {
     return gen(funRef);
   }
   template<typename A> M::Value *genval(const Ev::FunctionRef<A> &funRef) {
-    TODO();
+    TODO();  // Derived type functions (user + intrinsics)
+  }
+  template<Co::TypeCategory TC, int KIND>
+  M::Value *genval(const Ev::FunctionRef<Ev::Type<TC, KIND>> &funRef) {
+    if (const auto &intrinsic{funRef.proc().GetSpecificIntrinsic()}) {
+      std::string name{intrinsic->name};
```

Comment:
This is making a copy of the string, right? Do we need to do that?

Reply:
Correct, I had it like that because I was changing it in some earlier tests depending on the value but I should better change to a const ref for now.

Reply:
Changed to `llvm::StringRef` in https://github.com/schweitzpgi/f18/pull/3